debugger/debugcom.ml

```diff
@@ -95,24 +105,33 @@ let do_go_smallint n =
          | 'x' -> Exited
          | 's' -> Trap_barrier
          | 'u' -> Uncaught_exc
-         |  _  -> Misc.fatal_error "Debugcom.do_go" in
+         | 'D' -> Debug_info (input_value !conn.io_in :
+                                Instruct.debug_event list array)
+         | 'L' -> Code_loaded (input_binary_int !conn.io_in)
+         | 'U' -> Code_unloaded (input_binary_int !conn.io_in)
+         |  c  -> Misc.fatal_error (Printf.sprintf "Debugcom.do_go %c" c)
+       in
        let event_counter = input_binary_int !conn.io_in in
        let stack_pos = input_binary_int !conn.io_in in
-       let pc = input_binary_int !conn.io_in in
+       let frag = input_binary_int !conn.io_in in
+       let pos = input_binary_int !conn.io_in in
        { rep_type = summary;
-         rep_event_count = event_counter;
+         rep_event_count = Int64.of_int event_counter;
          rep_stack_pointer = stack_pos;
-         rep_program_pointer = pc })
+         rep_program_pointer = {frag; pos} })
 
 let rec do_go n =
   assert (n >= _0);
-  if n > max_small_int then(
-    ignore (do_go_smallint max_int);
-    do_go (n -- max_small_int)
-  )else(
+  if n > max_small_int then
+    begin match do_go_smallint max_int with
+    | { rep_type = Event } ->
+      do_go (n -- max_small_int)
+    | report ->
+      { report with
+        rep_event_count = report.rep_event_count ++ (n -- max_small_int) }
```

Comment:
I don't understand what this test is doing. Is there an explanation somewhere for why non-event values have to be handled differently? (Is this a bugfix?)

Reply:
The original patch from @whitequark is accompanied with a comment:

> As mentioned, this patch fixes some unrelated minor bugs:
>
>    in ocamldebug, if you passed a value over max_small_int to "step", then some breakpoints or exceptions could be ignored

I think this is indeed the bugfix: If the program has a non-`EVENT_COUNT` event (e.g., a breakpoint) during the first `max_small_int` steps, then this breakpoint should be reported to the user, instead of ignoring it.